blueprints/codes/formula.py

```diff
@@ -0,0 +1,62 @@
+"""Module for the abstract base class Formula."""
+from abc import ABC, abstractmethod
+
+
+class Formula(float, ABC):
+    """
+    Abstract base class for formulas used in the codes.
+    """
+
+    def __new__(cls, *args, **kwargs) -> "Formula":
+        """Method for creating a new instance of the class."""
+        result = cls.evaluate(*args, **kwargs)
+        return float.__new__(cls, result)
```

Comment:
I was just thinking! Could we use __call__ method instead of __new__ method? I know it would be a completely different approach, but it serves the same purpose of "not needing to call the evaluate method to get the result". I'm sure there will be downsides to that, but I can't think of one right now at the end of my work week! 😩 😉 
Please let me know if you need an example to be able to assess my suggestion.

Reply:
Switching from __new__ to __call__ in Formula means shifting from customizing object creation to defining behavior when an instance is used like a function. This changes usage to a two-step process (create, then call) and is unsuitable for immutable types like float, where value setup happens at instantiation. Essentially, __call__ makes instances function-like, diverging from the instantiation customization achieved with __new__, which is necessary for extending float.